### ISSUE TITLE
validateOne says array is valid even if one of the values in it is invalid

### ISSUE THREAD

**Issue by gibson** [aldeed/meteor-simple-schema#360]
i have a schema for user emails
```
    emails: {
        type: [Object],
        // this must be optional if you also use other login services like facebook,
        // but if you use only accounts-password, then it can be required
        optional: true,
        maxCount: 5
    },
    "emails.$.address": {
        type: String,
        regEx: SimpleSchema.RegEx.Email
    },
    "emails.$.verified": {
        type: Boolean
    }
```

And i try to validate this field by ```validateOne``` method
```ctx.validateOne({emails: [{address: 12321, verified: 'asdasd'}]}, 'emails’);```
On any value of address or verified i has answer ```true```

**Comment by abecks** [comment]
Fixed in PR #591

**Comment by aldeed** [comment]
Thanks for your patience. SimpleSchema 2.0.0-rc.1 is now released and should fix this bug.

There are a number of breaking changes when updating to 2.0, so be sure to check out [the change log](https://github.com/aldeed/meteor-simple-schema/blob/devel/CHANGELOG.md#200). If you use aldeed:collection2, you will need to use 2.10.0 or higher of that package in order to use SimpleSchema 2.0. If you use autoform, it is not yet updated to work with SimpleSchema 2.0, but hopefully soon.

SimpleSchema is now an isomorphic NPM package, so you can check out the updated readme and file issues over at the [other repo](https://github.com/aldeed/node-simple-schema). The Meteor wrapper package will exist for now but eventually I will probably deprecate support for it.

This is still a beta/RC and I do expect people will find issues, so use with caution. Production use is not yet recommended. That said, there are more and better unit tests than ever before, and the codebase should be much easier for others to read through and debug quickly.
